rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -0,0 +1,145 @@
+---
+Number: 0006
+Category: Standards Track
+Status: Draft
+Author: Ke Wang
+Organization: Nervos Foundation
+Created: 2018-12-01
+---
+
+# Complete Binary Merkle Tree
+
+## Introduction
+
+CKB uses Complete Binary Merkle Tree(CBMT) to generate Merkle Root and Merkle Proof for a static list of items, such as Transactions Root. Basically, CBMT is a complete binary tree as well as a full tree, it has the same node order with the complete binary tree. Compare with other Merkle trees, the hash computation of CBMT is minimal, as well as the proof size.
+
+## Tree Struct
+
+Tree with 6 items(the hashes are [T0, T1, T2, T3, T4, T5]) and tree with 7 items(the hashes are [T0, T1, T2, T3, T4, T5, T6]) is shown below:
+
+```
+        with 6 items                       with 7 items
+
+              B0                                 B0
+             /  \                               /  \
+           /      \                           /      \
+         /          \                       /          \
+       /              \                   /              \
+      B1              B2                 B1              B2
+     /  \            /  \               /  \            /  \
+    /    \          /    \             /    \          /    \
+   /      \        /      \           /      \        /      \  
+  B3      B4      TO      T1         B3      B4      B5      T0
+ /  \    /  \                       /  \    /  \    /  \
+T2  T3  T4  T5                     T1  T2  T3  T4  T5  T6
+```
+
+CBMT can be stored in an array, so the two trees above can be stored as:
```

Comment:
Need some description on the nodes ordering logic here (e.g. level by level, left to right)